kcqrs-appengine/src/main/kotlin/com/clouway/kcqrs/adapter/appengine/AppEngineKcqrs.kt

```diff
@@ -3,26 +3,51 @@ package com.clouway.kcqrs.adapter.appengine
 import com.clouway.kcqrs.core.*
 import com.clouway.kcqrs.core.messages.MessageFormatFactory
 
+
 /**
  * AppEngineKcqrs is an AppEngine implementation of Kcqrs which uses the GAE datastore for persistence of events
  * and TaskQueues for there handling.
  *
  * In most cases the events produced from commands may be published to messager broker like Pub-Sub by using the Intercetors
- * in the MessageBus. More information can be taken from Mess
+ * in the MessageBus.
  *
  * @author Miroslav Genov (miroslav.genov@clouway.com)
  */
-class AppEngineKcqrs(eventKind: String = "Event", eventHandlerEndpoint: String, messageFormatFactory: MessageFormatFactory) : Kcqrs {
-    private val messageBus = SimpleMessageBus()
-    private val eventStore = AppEngineEventStore(eventKind, messageFormatFactory.createMessageFormat())
-    private val eventPublisher = TaskQueueEventPublisher(eventHandlerEndpoint)
-    private val repository = EventRepository(eventStore, eventPublisher)
+class AppEngineKcqrs private constructor(private val messageBus: MessageBus,
+                     private val eventStore: EventStore,
+                     private val aggregateRepository: AggregateRepository) : Kcqrs {
+
+    override fun eventStore(): EventStore {
+        return eventStore
+    }
 
     override fun messageBus(): MessageBus {
         return messageBus
     }
 
-    override fun repository(): Repository {
-        return repository
+    override fun repository(): AggregateRepository {
+        return aggregateRepository
+    }
+
+    class Builder(private val configuration: Configuration, private val messageFormatFactory: MessageFormatFactory) {
+        private val messageBus = SimpleMessageBus()
+        var kind: String = "Event"
+        var kcqrsHandlerEndpoint = "/worker/kcqrs"
+        var queueName: String? = null
+        var identityProvider: IdentityProvider = IdentityProvider.Default()
+        var eventStore = AppEngineEventStore(kind, messageFormatFactory.createMessageFormat())
+        var eventPublisher: EventPublisher = TaskQueueEventPublisher(kcqrsHandlerEndpoint, queueName)
```

Comment:
I can't understad why we use TaskQueueEventPublisher by default

Reply:
Do you have something about it ? I hear from you several times that TaskQueue shouldn't be used by I didn't understood why.

Reply:
I'm not sayning that we should NOT use it, In the example the events are transported throught this emssage bus (taskqueueu) - async. 
Nevermind, will see when we start implementing the flow...